File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/anoncreds/IssuerCreateCredentialTest.java
package org.hyperledger.indy.sdk.anoncreds;

import org.hyperledger.indy.sdk.InvalidStructureException;
import org.junit.*;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;

public class IssuerCreateCredentialTest extends AnoncredsIntegrationTest {

	@Test
	public void testIssuerCreateCredentialWorks() throws Exception {}

	@Test
	public void testIssuerCreateCredentialWorksForInvalidCredentialValues() throws Exception {

		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(InvalidStructureException.class));

		String credValues = "{" +
				"        \"sex\":\"male\",\n" +
				"        \"age\":\"28\"" +
				"       }";

		Anoncreds.issuerCreateCredential(wallet, issuer1GvtCredOffer, issuer1GvtCredReq, credValues, null, - 1).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/AnonCredsTests/IssuerCreateCredentialTest.cs
using Hyperledger.Indy.AnonCredsApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.AnonCredsTests
{
    [TestClass]
    public class IssuerCreateCredentialTest : AnonCredsIntegrationTestBase
    {
        [TestMethod]
        public async Task TestIssuerCreateAndStoreCredentialWorks()
        {
        }

        [TestMethod]
        public async Task TestIssuerCreateAndStoreCredentialWorksForCredentialValuesDoNotCorrespondToCredentialRequest()
        {
            var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
                AnonCreds.IssuerCreateCredentialAsync(wallet, issuer1GvtCredOffer, issuer1GvtCredReq, xyzCredentialValuesJson, null, null)
            );
        }

        [TestMethod]
        public async Task TestIssuerCreateAndStoreCredentialReqWorksForInvalidCredentialValues()
        {
            var credentialValues = "{\"sex\":\"male\",\n" +
                    "        \"age\":\"28\"" +
                    "       }";

            var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
                AnonCreds.IssuerCreateCredentialAsync(wallet, issuer1GvtCredOffer, issuer1GvtCredReq, credentialValues, null, null)
            );
        }
    }
}
